src/ansys/fluent/core/launcher/launcher.py

```diff
@@ -764,162 +982,12 @@ def launch_fluent(
-                        file_type="case-data", file_name=case_data_file_name
-                    )
-                else:
-                    raise RuntimeError(
-                        "Case and data file cannot be read in meshing mode."
-                    )
-
-            return session
-        except Exception as ex:
-            logger.error(f"Exception caught - {type(ex).__name__}: {ex}")
-            raise LaunchFluentError(launch_cmd) from ex
-        finally:
-            server_info_file = Path(server_info_file_name)
-            if server_info_file.exists():
-                server_info_file.unlink()
-    elif fluent_launch_mode == LaunchMode.PIM:
-        logger.info(
-            "Starting Fluent remotely. The startup configuration will be ignored."
-        )
-
-        if product_version:
-            fluent_product_version = "".join(product_version.split("."))[:-1]
-        else:
-            fluent_product_version = None
-
-        return launch_remote_fluent(
-            session_cls=new_session,
-            start_transcript=start_transcript,
-            product_version=fluent_product_version,
-            cleanup_on_exit=cleanup_on_exit,
-            meshing_mode=meshing_mode,
-            dimensionality=version,
-            launcher_args=argvals,
-        )
-
-    elif fluent_launch_mode == LaunchMode.CONTAINER:
-        args = _build_fluent_launch_args_string(**argvals).split()
-        if meshing_mode:
-            args.append(" -meshing")
-
-        if dry_run:
-            if container_dict is None:
-                container_dict = {}
-            config_dict, *_ = configure_container_dict(args, **container_dict)
-            from pprint import pprint
-
-            print("\nDocker container run configuration:\n")
-            print("config_dict = ")
-            if os.getenv("PYFLUENT_HIDE_LOG_SECRETS") != "1":
-                pprint(config_dict)
-            else:
-                config_dict_h = config_dict.copy()
-                config_dict_h.pop("environment")
-                pprint(config_dict_h)
-                del config_dict_h
-            return config_dict
-
-        port, password = start_fluent_container(args, container_dict)
-
-        session = new_session(
-            fluent_connection=FluentConnection(
-                port=port,
-                password=password,
-                cleanup_on_exit=cleanup_on_exit,
-                start_transcript=start_transcript,
-                launcher_args=argvals,
-                inside_container=True,
-            ),
-            remote_file_handler=RemoteFileHandler(),
-        )
-
-        if start_watchdog is None and cleanup_on_exit:
-            start_watchdog = True
-        if start_watchdog:
-            logger.debug("Launching Watchdog for Fluent container...")
-            watchdog.launch(os.getpid(), port, password)
-
-        return session
+    launcher = create_launcher(argvals.pop("fluent_launch_mode"))
```

Comment:
you can do a `del fluent_launch_mode` before calling `argvals = locals().copy()`

Reply:
We want to remove "fluent_launch_mode" and pass it's value to `create_launcher` at the same time. We can do this using `pop` not `del`. Therefore we are using `pop` in `launcher = create_launcher(argvals.pop("fluent_launch_mode"))`.